indico/modules/events/sessions/clone.py

```diff
@@ -51,30 +58,43 @@ def _clone_sessions(self, new_event):
                           subqueryload('acl_entries')))
         for old_sess in query:
             sess = Session()
-            sess.populate_from_attrs(old_sess, attrs)
-            sess.blocks = list(self._clone_session_blocks(old_sess.blocks))
+            if event_exists:
+                sess.populate_from_attrs(old_sess, attrs, skip='friendly_id')
+                sess.friendly_id = self._get_next_friendly_id(new_event.id)
+            else:
+                sess.populate_from_attrs(old_sess, attrs)
+            sess.blocks = list(self._clone_session_blocks(old_sess.blocks, event_exists=event_exists))
             sess.acl_entries = clone_principals(SessionPrincipal, old_sess.acl_entries,
                                                 self._event_role_map, self._regform_map)
             new_event.sessions.append(sess)
             self._session_map[old_sess] = sess
 
-    def _clone_session_blocks(self, blocks):
+    def _clone_session_blocks(self, blocks, event_exists=False):
         attrs = get_simple_column_attrs(SessionBlock) | {'own_room', 'own_venue'}
         for old_block in blocks:
             block = SessionBlock()
             block.populate_from_attrs(old_block, attrs)
-            block.person_links = list(self._clone_person_links(old_block.person_links))
+            block.person_links = list(self._clone_person_links(old_block.person_links,
+                                                               event_exists=event_exists))
             self._session_block_map[old_block] = block
             yield block
 
-    def _clone_person_links(self, person_links):
+    def _clone_person_links(self, person_links, event_exists=False):
         attrs = get_simple_column_attrs(SessionBlockPersonLink)
         for old_link in person_links:
             link = SessionBlockPersonLink()
             link.populate_from_attrs(old_link, attrs)
-            link.person = self._person_map[old_link.person]
+            current_person = self._person_map[old_link.person]
+            if event_exists and current_person.user_id in self._new_event_persons.keys():
```

Comment:
```suggestion
            if event_exists and current_person.user_id in self._new_event_persons:
```

no need for `.keys()` to check if a key exists